src/python/pants/backend/native/config/environment.py

```diff
@@ -166,15 +163,95 @@ def get_invocation_environment_dict(self, platform):
 class GCCCCompiler(datatype([('c_compiler', CCompiler)])): pass
 
 
-class LLVMCCompiler(datatype([('c_compiler', CCompiler)])): pass
+class GCCCLinker(datatype([('c_linker', Linker)])): pass
 
 
 class GCCCppCompiler(datatype([('cpp_compiler', CppCompiler)])): pass
 
 
+class GCCCppLinker(datatype([('cpp_linker', Linker)])): pass
+
+
+class LLVMCCompiler(datatype([('c_compiler', CCompiler)])): pass
+
+
+class LLVMCLinker(datatype([('c_linker', Linker)])): pass
+
+
 class LLVMCppCompiler(datatype([('cpp_compiler', CppCompiler)])): pass
 
 
+class LLVMCppLinker(datatype([('cpp_linker', Linker)])): pass
+
+
+class CToolchain(datatype([('c_compiler', CCompiler), ('c_linker', Linker)])): pass
+
+
+class CToolchainProvider(object):
+
+  @abstractproperty
+  def as_c_toolchain(self):
+    """???"""
```

Comment:
Please doc or pass.

Reply:
I think this should just be a `pass`, will make that change.

Reply:
Removed in 3441e74!